lib/Doctrine/Annotations/Metadata/AnnotationMetadata.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Metadata;
+
+use function array_combine;
+use function array_filter;
+use function array_map;
+use function array_values;
+use function assert;
+use function count;
+
+/**
+ * @internal
+ */
+final class AnnotationMetadata
+{
+    /** @var string */
+    private $name;
+
+    /** @var AnnotationTarget */
+    private $target;
+
+    /** @var bool */
+    private $usesConstructor;
+
+    /** @var PropertyMetadata[] */
+    private $properties;
+
+    /** @var PropertyMetadata|null */
+    private $defaultProperty;
+
+    /**
+     * @param PropertyMetadata[] $properties
```

Comment:
This is not necessary as the variadic argument already contains a proper type hint.